pcap/stream.go

```diff
@@ -71,9 +85,18 @@ func newTCPFlow(clock clockWrapper, bidiID akinet.TCPBidiID, nf, tf gopacket.Flo
 		bidiID:          bidiID,
 		outChan:         outChan,
 		factorySelector: fs,
+		firstPacketSeen: false,
 	}
 }
 
+func (f *tcpFlow) FirstPacketSeen() bool {
+	return f.firstPacketSeen
+}
+
+func (f *tcpFlow) TcpFlowInitiator() tcpFlowInitiator {
+	return *f.initiator
+}
```

Comment:
This is one of the reasons to worry about using *T instead of optional[T], the dereference here is unchecked.

(And might be the nil pointer derefs you are getting in unit tests?  I didn't investigate.)